clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -16542,22 +16542,75 @@ Value *CodeGenFunction::EmitPPCBuiltinExpr(unsigned BuiltinID,
 
   Intrinsic::ID ID = Intrinsic::not_intrinsic;
 
+#include "llvm/TargetParser/PPCTargetParser.def"
```

Comment:
This is getting a bit tangled. Can you please provide two static functions:
`EmitPPCAIXCpuIDFunc(...), EmitPPCLinuxCpuIDFunc(...)` and just call the right one from here.